apps/web/src/pages/integrations/components/multi-provider/UpdateProviderSidebar.tsx

```diff
@@ -167,22 +167,26 @@ export function UpdateProviderSidebar({
       return;
     }
 
-    const { channel: selectedChannel, environmentId, primary } = selectedProvider;
+    const { channel: selectedChannel, environmentId, primary, conditions } = selectedProvider;
     const isActiveFieldChanged = dirtyFields.active;
     const hasSameChannelActiveIntegration = !!providers
       .filter((el) => !NOVU_PROVIDERS.includes(el.providerId) && el.integrationId !== selectedProvider.integrationId)
       .find((el) => el.active && el.channel === selectedChannel && el.environmentId === environmentId);
     const isChannelSupportPrimary = CHANNELS_WITH_PRIMARY.includes(selectedChannel);
+    const hasAddedCondition = primary && conditions && conditions.length > 0; // show modal
+    const hasNotAddedConditionOnActive = !(isActive && conditions && conditions.length > 0); // show modal
 
     if (
       isActiveFieldChanged &&
       isChannelSupportPrimary &&
+      hasAddedCondition &&
+      hasNotAddedConditionOnActive &&
```

Comment:
I feel like this should be updated. Basically, you want to show the modal only when the current integration is primary and has conditions applied (no matter the active flag). So it will be like:
```
const isChangedToActive =  isActiveFieldChanged && isChannelSupportPrimary && isActive && hasSameChannelActiveIntegration;

const isChangedToInactiveAndIsPrimary = isActiveFieldChanged && isChannelSupportPrimary && !isActive && primary && hasSameChannelActiveIntegration;

const isPrimaryAndHasConditions = primary && conditions && conditions.length > 0;

if (isChangedToActive || isChangedToInactiveAndIsPrimary || isPrimaryAndHasConditions)
```